gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/ResumeDagProc.java

```diff
@@ -82,12 +82,14 @@ protected void act(DagManagementStateStore dagManagementStateStore, Optional<Dag
         Map<String, String> jobMetadata = TimingEventUtils.getJobMetadata(Maps.newHashMap(), node.getValue());
         eventSubmitter.getTimingEvent(TimingEvent.LauncherTimings.JOB_PENDING_RESUME).stop(jobMetadata);
       }
-      // Set flowStartTime so that flow SLA will be based on current time instead of original flow
+      // Set flowStartTime so that flow start deadline and flow completion deadline will be based on current time instead of original flow
       node.getValue().setFlowStartTime(flowResumeTime);
     }
 
+    // these two statements effectively move the dag from failed dag store to (running) dag store
```

Comment:
maybe add - "to prevent loss in the unlikely event of failure between the two, we add first"

that said, should we check for this state around L68/69?

Reply:
check for what state?

Reply:
check for whether we do find a failed dag, but the same one is also in the dag state store (so presumably a prior exec had failed between line 90 and 91)